src/main/java/org/ohdsi/usagi/ui/ImportDialog.java

```diff
@@ -450,11 +455,13 @@ public void run() {
 				})).get();
 				forkJoinPool.shutdown();
 				dialog.setVisible(false);
-				Global.applyPreviousMappingAction.setEnabled(true);
-				Global.saveAction.setEnabled(true);
-				Global.saveAsAction.setEnabled(true);
-				Global.exportAction.setEnabled(true);
-				Global.exportForReviewAction.setEnabled(true);
+				synchronized (dialog) {
+					Global.applyPreviousMappingAction.setEnabled(true);
+					Global.saveAction.setEnabled(true);
+					Global.saveAsAction.setEnabled(true);
+					Global.exportAction.setEnabled(true);
+					Global.exportForReviewAction.setEnabled(true);
+				}

```

Comment:
If this is supposed to be globally synced, it may be better to delegate the syncing to global. E.g.
```java
Global.edit(() -> {
    Global.applyPreviousMappingAction.setEnabled(true);
    Global.saveAction.setEnabled(true);
    Global.saveAsAction.setEnabled(true);
    Global.exportAction.setEnabled(true);
    Global.exportForReviewAction.setEnabled(true);
});
```
and then Global.edit accepts a `Runnable` and calls
```java
private static final Object lockObject = new Object();

public void edit(Runnable block) {
synchronized (lockObject) {
    block.run();
}
}
```

Even nicer would be that there was a settings object inside global and synchronise on that, but that would be a step to far for this PR.

Reply:
This is odd... I have experimented with synchronized elsewhere, to no avail. I was not even aware that I did this? I will remove it.